Fix error with empty style definitions json file

EasterPeanut opened the pull request:
An artifact in our own codebase could sometimes cause empty style definitions json files. This will fail compilation of ex_css_modules because decoding an empty string is not working.

To capture this edge case, let's check if the file is empty before trying to decode OR generate the json file through a task.

sn3p commented on lib/ex_css_modules/ex_css_modules.ex:
```diff
@@ -37,21 +37,40 @@ defmodule ExCSSModules do
 
-      true ->
-        %{}
+        not is_nil(build_json_task) ->
```
Match for task/module here?

sn3p commented on lib/ex_css_modules/ex_css_modules.ex:
```diff
@@ -37,24 +37,45 @@ defmodule ExCSSModules do
+
+          raise(CompileError, description: message, file: "#{filename}.json")
+
+        empty_file?(filename) ->
```
```suggestion
        empty_file?("#{filename}.json") ->
```

no?

EasterPeanut commented on README.md:
```diff
@@ -28,7 +32,18 @@ defmodule Mix.Tasks.GenerateMissingJson do
+use ExCSSModules.View,
+  namespace: MyApplicationWeb,
+  embed_stylesheet: ApplicationSettings.built_for?(:prod),
+  build_stylesheet_definitions_json_task: unquote(Mix.Tasks.BuildScopedJson),
```
```suggestion
  build_stylesheet_definitions_json_task: unquote(Mix.Tasks.GenerateMissingJson),
```

zwippie commented:
Tested in detroit with this branch. Added `build_stylesheet_definitions_json_task` option in web.ex. Deleted a random style.css json file. Results was a `(UndefinedFunctionError) function Mix.Tasks.GenerateMissingJson.run/1 is undefined` but I guess that task is on its way.

Nice!

EasterPeanut commented:
> Tested in detroit with this branch. Added `build_stylesheet_definitions_json_task` option in web.ex. Deleted a random style.css json file. Results was a `(UndefinedFunctionError) function Mix.Tasks.GenerateMissingJson.run/1 is undefined` but I guess that task is on its way.
> 
> 
> 
> Nice! 

Yes! That will be done in https://github.com/DefactoSoftware/detroit/pull/13193 :D
